internal/server/data/destination.go

```diff
@@ -67,12 +68,92 @@ func UpdateDestination(tx WriteTxn, destination *models.Destination) error {
 	return update(tx, (*destinationsUpdateTable)(destination))
 }
 
-func GetDestination(db GormTxn, selectors ...SelectorFunc) (*models.Destination, error) {
-	return get[models.Destination](db, selectors...)
+type GetDestinationOptions struct {
+	// ByID instructs GetDestination to return the row matching this ID. When
+	// this value is set, all other fields on this strut will be ignored
+	ByID uid.ID
+	// ByUniqueID instructs GetDestination to return the row matching this
+	// uniqueID.
+	ByUniqueID string
+	// ByName instructs GetDestination to return the row matching this name.
+	ByName string
```

Comment:
I added this now that we have a unique constraint on the name field. There's a test case for it as well